fix: update trivy-action to 0.34.1 (v0.57.1 binary no longer available)

## Summary

`docker-publish.yml` の `aquasecurity/trivy-action@0.29.0` が Trivy `v0.57.1` バイナリを要求するが、そのリリースは GitHub Releases から削除されており、scan ジョブが失敗する。

## 再現

`v2.0.0` タグ push 時に Docker Publish ワークフローの scan ジョブが失敗:
```
aquasecurity/trivy info checking GitHub for tag 'v0.57.1'
aquasecurity/trivy info found version: 0.57.1 for v0.57.1/Linux/64bit
##[error]Process completed with exit code 1.
```

Run: https://github.com/Route-Reflector/simnos/actions/runs/22565944830

## Root Cause

- `trivy-action@0.29.0` がデフォルトで `v0.57.1` を使用
- Trivy `v0.57.1` は GitHub Releases に存在しない（最新は `v0.69.2`）

## Fix

`aquasecurity/trivy-action` を `0.29.0` → `0.34.1` にバンプする。

--- a/.github/workflows/docker-publish.yml
+++ b/.github/workflows/docker-publish.yml
@@ -58,7 +58,7 @@ jobs:
           cache-from: "type=gha,scope=docker-scan"
           cache-to: "type=gha,mode=max,scope=docker-scan"
       - name: "Trivy vulnerability scan"
-        uses: "aquasecurity/trivy-action@0.29.0"
+        uses: "aquasecurity/trivy-action@0.34.1"
         with:
           image-ref: "${{ env.IMAGE_NAME }}:scan"
           format: "table"

